pkg/cmd/version/version.go

```diff
@@ -48,19 +67,51 @@ func NewCmdVersion() (*cobra.Command, *Options) {
 }
 
 // Run implements the command
-func (o *Options) Run() error {
+func (o *Options) Run(cmd *cobra.Command) error {
```

Comment:
Instead of passing cobra.Command, just pass the io writer?
Example: https://github.com/helm/helm/blob/b1e24764325135b364a7f81fb0aab683d190c546/cmd/helm/version.go#L82

Reply:
This will cause the change of the root.go file as it needs to pass the buffer to it also?
https://github.com/jenkins-x/jx/blob/88d12dd14c8cb7ea0afd66a63e6bc6c425776519/pkg/cmd/root.go#L62

Reply:
That should be fine imo.

Reply:
or u can pass the writer as a field in the options struct?

Reply:
How would this affect the root.go file? Should I just create the buffer and Pass it to NewCmdVersion() and the output will be printed through fmt.Fprintf() inside Run()?

Reply:
It wont change root.go.
You can change the options struct in version package:
```
type Options struct {
	Verbose bool
	Quiet   bool
	Out     io.Writer
}
```

and in Run function, check if o.Out is nil, if it's nil set it to os.Stdout. In your tests, you can set o.Out to a buffer and go with ur checks.

For printing you can use `fmt.Fprintf`

Reply:
I think it is done now